engines/scumm/players/player_nes.cpp

```diff
@@ -361,9 +359,8 @@ static const uint32 NoiseFreq[16] = {
 };
 
 void Noise::Reset() {
-	memset(this, 0, sizeof(*this));
+	*this = Noise();
 	CurD = 1;
-	Cycles = 1;
 	EnvCtr = 1;
```

Comment:
These two should also be initialized with appropriate default values, like `Cycles`